src/views/PickupLocationModal.vue

```diff
@@ -0,0 +1,182 @@
+      facilityId: ''
+    }
+  },
+  props: ["shipGroup", "productIds"],
+  mounted() {
+    this.getNearbyStores()
+  },
+  methods: {
+    async getNearbyStores() {
+      try {
+        let shipGroupLocationResp: any, productInventoryResp: any, storeLookupResp: any, facilityIds;
+        if (this.shipGroup.shipmentMethodTypeId === 'STOREPICKUP' && this.shipGroup.selectedShipmentMethodTypeId === 'STOREPICKUP') {
+          storeLookupResp = await FacilityService.getStores({
+            "viewSize": 60,
+            "filters": ["storeType: RETAIL_STORE"],
+          })
+
+          if (storeLookupResp.status !== 200 || hasError(storeLookupResp) || !storeLookupResp.data.response.numFound) {
+            return showToast(translate("Something went wrong while fetching nearby stores"))
+          }
+        } else {
+          shipGroupLocationResp = await FacilityService.getLocation({
+            "json": {
+              "query": `postcode: ${this.shipGroup.shipTo.postalAddress.postalCode}`
+            }
+          })
+
+          if (shipGroupLocationResp.status !== 200 || hasError(shipGroupLocationResp) || !shipGroupLocationResp.data.response.numFound) {
+            return showToast(translate("Something went wrong while fetching nearby stores"))
+          }
+
+          storeLookupResp = await FacilityService.getStores({
+            "viewSize": 60,
+            "filters": ["storeType: RETAIL_STORE"],
+            "point": shipGroupLocationResp.data.response.docs[0].location,
+            "distance": 50,
+            "fieldsToSelect": ["storeCode", "storeName", "dist"]
+          })
+
+          if (storeLookupResp.status !== 200 || hasError(storeLookupResp) || !storeLookupResp.data.response.numFound) {
+            return showToast(translate("Something went wrong while fetching nearby stores"))
+          }
+        }
+        facilityIds = storeLookupResp.data.response.docs.map((store: any) => store.storeCode)
+        productInventoryResp = await StockService.checkInventory({
+          "filters": {
+            "productId": this.productIds,
+            "facilityId": facilityIds
+          },
+          "fieldsToSelect": ["atp", "facilityName", "facilityId"],
+        });
+
+        if (productInventoryResp.status !== 200 || hasError(productInventoryResp) || !productInventoryResp.data.count) {
+          return showToast(translate("Something went wrong while fetching nearby stores"))
+        }
+        const storesWithInventory = productInventoryResp.data.docs.filter((store: any) => store.atp > 0)
+        storesWithInventory.map((store: any) => {
+          const storeDetails = storeLookupResp.data.response.docs.find((data: any) => data.storeCode === store.facilityId );
+          if (storeDetails) this.nearbyStores.push({...store, distance: storeDetails.dist});
+        });
+        // sorting in alphabetical order 
+        this.nearbyStores.sort((a: any, b: any) => a.facilityName.localeCompare(b.facilityName))
```

Comment:
I think there is no need to sort the stores explicitly, as the stores are be default sorted alphabetically by storeName.

Reply:
Updated the looping